docs/proposals/vision.md

```diff
@@ -0,0 +1,130 @@
+in non-Kubernetes clusters. They wrote Heapster in a such a way that metrics can be read not 
+only from Kubernetes nodes (via Kubelet API) but also from custom deployments via cAdvisor 
+(with support for CoreOS Fleet and flat file node lists). 
+
+Metrics collected by Heapster can be written into multiple kinds of storage - Influxdb, 
+OpenTSDB, Google Cloud Monitoring, Hawkular, Kaflka, Riemann, ElasticSearch (some of them are
+not yet submitted).
+
+In addition to gathering metrics  Heapster is responsible for handling Kubernetes events - it 
+reads them from Kubernetes API server and writes them, without extra processing, to a selection
+of persistent storages: Google Cloud Logging, Influxdb, Kafka, OpenTSDB, Hawkular, 
+ElasticSearch, etc.
+
+There is/was a plan to add resource prediction components (Initial Resources, Vertical 
+Pod Autoscaling) to Heapster binary.
+
+## Separation of Use Cases
+From the current state description (see above) the following use cases can be extracted:
+
+* [UC1] Read metrics from nodes and write them to an external storage.
+* [UC2] Expose metrics from the last 2-3 minutes (for HPA and GKE)
+* [UC3] Read Events from the API server and write them to a permanent storage
+* [UC4] Do some long-term (hours, days) metrics analysis to get stats (average, 95 percentile) 
+and expected resource usage.
+* [UC5] Provide cpu and memory metrics for longer time window for the new Kubernetes 
+Dashboard UI (15 min for 1.2, up to 1h later for plots) 
+
+UC1 and UC2 go together - to expose the most recent metrics the API should be connected 
+to the metrics stream.
+UC3 can be completely separated from UC1, UC2 and UC4 - it reads different data from a 
+different place and writes it in a slightly different format to different sinks. 
+UC4 is connected to UC1 and UC2 but it is more based on data from the permanent storage 
+than on the super-fresh metrics stored in the memory.
+UC5 can go either with UC1/UC2 or with UC4. As there is no immediate need for UC4 we will
+ provide basic UC5 together with UC1/UC2 but in the future it will join UC4.
+
+This separation leads to an idea of splitting Heapster into 3 binaries:
+
+* Core Heapster - covering UC1, UC2 and temporarily UC5
+* Eventer - covering UC3
+* Oldtimer - covering UC4 and UC5 
+
+## Reduction of Responsibility
+
+With 3 possible node sources (Kuberentes API Server, flat file, CoreOS Fleet), 2 metrics 
+sources (cAdvisor and Kubelet) and constantly growing number of sinks we have to separate 
+the stuff that the core Heapster/K8S team is responsible for and what is provided as a 
+plugin/addition and doesn’t come in the main release package. 
+
+We decided to focus only on:
+
+Kubernetes API Server node source
+Kubelet metrics source
+Influxdb, GCM, GKE (there is special endpoint for GKE that exposes all available metrics),
+ Hawkular sinks for Heapster
+Influxdb, GCL sinks for Eventer
```

Comment:
Kafka is not a "trivial" deployment and anyone who has it will probably be able to build his own image. With 1.2 the sinks will remain compiled in. We can work out some specific requirements later (tests, e2e tests, performance checks, only "official" client libraries etc).